### PR TITLE
Upgrade to OpenLayers v7.x

### PR DESCRIPTION
**Pull request by annarieger**
### Breaking change

- Bump `ol` dependency and peer dependency to v7.x
- Bump `ol-util` dependency to v10.x, which also relies on the OpenLayers 7

The PR also contains some required adaptations due to major upgrade of the mentioned libraries:

- The deprecated method `map.forEachLayerAtPixel` was replaced by `layer.getData()`
  - s. [OpenLayers release notes v7.0.0](https://github.com/openlayers/openlayers/releases/tag/v7.0.0) for further details
- Fixed wrong typings and add missed method arguments to some methods of `ol-util` classes
- Fixed some linting warnings
- Added some further null checks

Please review @terrestris/devs 

## Pull request type

- [x] Bugfix
- [ ] Feature
- [x] Dependency updates
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

## Do you introduce a breaking change?

- [x] Yes
- [ ] No

## Checklist

- [x] I understand and agree that the changes in this PR will be licensed under the
  [BSD-2-Clause](https://github.com/terrestris/react-geo/blob/main/LICENSE).
- [x] I have followed the [guidelines for contributing](https://github.com/terrestris/react-geo/blob/main/CONTRIBUTING.md).
- [x] The proposed change fits to the content of the [Code of Conduct](https://github.com/terrestris/react-geo/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added or updated tests and documentation, and the test suite passes (run `npm test` locally).
- [ ] I have added a screenshot/screencast to illustrate the visual output of my update.

### CODE REVIEW

File: src/Button/CopyButton/CopyButton.spec.tsx [hunk 1]
```diff
@@ -16,8 +19,19 @@ describe('<CopyButton />', () => {
   const coord = [829729, 6708850];
   let map: OlMap;
   let feature: OlFeature<OlPoint>;
+  let style: OlStyleStyle;
 
   beforeEach(() => {
+
```
Comment (dnlkoch) [inline src/Button/CopyButton/CopyButton.spec.tsx:25]:
```suggestion
```

File: src/Button/ToggleGroup/ToggleGroup.tsx [hunk 2]
```diff
@@ -153,6 +153,7 @@ class ToggleGroup extends React.Component<ToggleGroupProps, ToggleGroupState> {
       if (React.isValidElement(child)) {
         // pass the press state through to child components
         return React.cloneElement(child, {
+          // @ts-ignore
```
Comment (dnlkoch) [inline src/Button/ToggleGroup/ToggleGroup.tsx:156]: Any chance to get this one fixed?

**Review by dnlkoch — APPROVED** [review]
Awesome @annarieger! Thanks!

File: src/Button/ToggleGroup/ToggleGroup.tsx (hunk 2 above)
Comment (annarieger) [reply src/Button/ToggleGroup/ToggleGroup.tsx]: Got it with eabfd17

**Pull request merged**